### PR TITLE
Wx handle error restart

### PR DESCRIPTION
**Pull request by le-horizon**
[accompanying hobot change](https://github.com/HorizonRoboticsInternal/Hobot/pull/1498)

### CODE REVIEW

File: wx_armor/wx_armor/wx_armor_ws.cc [hunk 1]
```diff
@@ -57,43 +51,53 @@ void WxArmorWebController::handleNewMessage(const WebSocketConnectionPtr &conn,
...
-      ClientState &state = conn->getContextRef<ClientState>();
-      state.engaging = true;
-      state.latest_healthy_time = std::chrono::system_clock::now();
+    // Relay the command to the driver.
+    nlohmann::json json = nlohmann::json::parse(payload);
+    std::vector<float> position(json.size());
+    for (size_t i = 0; i < json.size(); ++i) {
+      position[i] = json.at(i).get<float>();
+    }
+    CheckAndSetPosition(position, 0.0);
```
Comment (QuantuMope) [inline wx_armor/wx_armor/wx_armor_ws.cc:77]: Still unsure of what the effects of setting `moving_time = 0`.

**Review by QuantuMope — COMMENTED** [review]
Hey Le, just had some minor comments. Otherwise looks solid to me. Thanks for the changes!

File: wx_armor/wx_armor/wx_armor_ws.cc [hunk 2]
```diff
@@ -128,6 +132,8 @@ void SlowDownToStop(const SensorData &curr_reading,
 
   // Overwrite the current trajectory with our decelerating one.
   Driver()->SetPosition(targets, deceleration_time, 0.49 * deceleration_time);
+  // SetPID to a small Kp, large Kd and zero Ki to drop to the groundfrom current pos.
+  Driver()->SetPID({{"all", 0, 0, 80000}});
```
[the remark this answers is not in the document]
Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_ws.cc]: Done

File: wx_armor/wx_armor/wx_armor_driver.h [hunk 3]
```diff
@@ -162,6 +163,8 @@ class WxArmorDriver {
    */
   std::vector<float> GetSafetyVelocityLimits();
 
+  std::vector<float> GetSafetyCurrentLimits();
```
[the remark this answers is not in the document]
Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_driver.h:178]: Done

**Review by le-horizon — COMMENTED** [review]
Done or addressed all.  PTAL.  Thanks,

Le

File: wx_armor/wx_armor/wx_armor_ws.cc [hunk 4]
```diff
@@ -185,9 +185,9 @@ WxArmorWebController::GuardianThread::GuardianThread() {
       if (sensor_data.has_value() || Driver()->SafetyViolationTriggered()) {
         if (!sensor_data.has_value()) {
           // If we have an error, we still want to publish the error codes
-          sensor_data = SensorData{.pos = std::vector<float>{},
-                                   .vel = std::vector<float>{},
-                                   .crt = std::vector<float>{},
+          sensor_data = SensorData{.pos = std::vector<float>{0,0,0,0,0,0,0},
+                                   .vel = std::vector<float>{0,0,0,0,0,0,0},
+                                   .crt = std::vector<float>{0,0,0,0,0,0,0},
```
Comment (QuantuMope) [inline wx_armor/wx_armor/wx_armor_ws.cc:190]: We can change this to `std::vector<float>(num_joints)`.

**Review by QuantuMope — COMMENTED** [review]
Hey Le, just had one minor comment.

File: wx_armor/wx_armor/wx_armor_ws.cc (hunk 4 above)
Comment (le-horizon) [reply wx_armor/wx_armor/wx_armor_ws.cc]: Good point.  Done.  PTAL.

**Review by QuantuMope — APPROVED** [review]
LGTM

**Pull request merged**